automates/program_analysis/CAST2GrFN/model/cast_to_air_model.py

```diff
@@ -0,0 +1,417 @@
+from typing import List, Dict
+from enum import Enum
+
+from automates.program_analysis.CAST2GrFN.model.cast import AstNode
+
+
+class CASTToAIRException(Exception):
```

Comment:
@dpdicken lets stick with the `obj2obj` naming scheme here and change the class name to: `C2A<error-type>`.

Additionally, I would like to break down this single exception into a family of exceptions. Minimally I would like the following to be included:
- `C2ATypeError`: inherits from TypeError and is used for any type errors that occur during CAST --> AIR processing (see `cast_to_air_visitor.py::101`)
- `C2ARuntimeError`: inherits from RuntimeError and is used for any runtime errors that occur during CAST --> AIR processing
- `C2ANameError`: inherits from NameError and is used when name errors occur (such as a missing member variable for some object) during CAST --> AIR processing
- `C2AValueError`: inherits from ValueError and is used when an operation cannot be performed for a given value during CAST --> AIR processing